ecomap/api/v1_0/models.py

```diff
@@ -1,169 +1,205 @@
-from sqlalchemy import Column, Enum, Integer, String, Text, text, create_engine, ForeignKey, DateTime
+from sqlalchemy import Column, Enum, Integer, String, Text, ForeignKey, \
+    DateTime, Table
+
+from sqlalchemy.orm import relationship
+
 from sqlalchemy.dialects.postgresql import JSON
 from geoalchemy2 import Geometry
 from sqlalchemy.ext.declarative import declarative_base
-from api import settings
-from api.utils.db import get_db_engine
-
 
-# Turn off engine echo setting
-settings['debug'] = False
-engine = get_db_engine(settings)
 Base = declarative_base()
 
-
 enum_status = ('SOLVED', 'UNSOLVED')
 enum_activity_type = ('ADDED', 'REMOVED', 'UPDATED')
 enum_severity_type = ('1', '2', '3', '4', '5')
 enum_modifiers = ('ANY', 'OWN', 'NONE')
 enum_actions = ('GET', 'PUT', 'POST', 'DELETE')
 
+role_permissions = Table(
+    'role_permissions', Base.metadata,
+    Column('role', Integer, ForeignKey('roles.id'), primary_key=True,
+           nullable=False),
+    Column('permission', Integer, ForeignKey('permissions.id'),
+           primary_key=True,
+           nullable=False)
+)
+
+user_roles = Table(
+    'user_roles', Base.metadata,
+    Column('user', Integer, ForeignKey('users.id'), primary_key=True,
+           nullable=False),
+    Column('role', Integer, ForeignKey('roles.id'), primary_key=True,
+           nullable=False)
+)
+
 
 class ProblemType(Base):
     __tablename__ = 'problem_types'
 
     id = Column(Integer, primary_key=True)
-    type = Column(String(100), nullable=False)
+    type = Column(String, nullable=False)
 
 
 class Role(Base):
     __tablename__ = 'roles'
 
     id = Column(Integer, primary_key=True)
```

Comment:
are you sure? it works fine for me.
just check and if it doesn't work for you, i'll add autoincrements.